noxfile.py

```diff
@@ -0,0 +1,149 @@
+from pathlib import Path
+from shutil import rmtree
+from subprocess import PIPE, run
+
+import nox
+
+nox.options.sessions = ["clean", "build", "open"]
+
+PROJECT_ROOT = Path(__file__).parent
+DOC = PROJECT_ROOT / "doc"
+DOC_SCRIPTS = DOC / "scripts"
+DOC_BUILD = DOC / "build"
+PACKAGE = PROJECT_ROOT / "exasol_sphinx_github_pages_generator"
+ENTRY_POINT = "exasol_sphinx_github_pages_generator.deploy_github_pages"
+
+
+def _tags():
+    """
+    Returns a list of all tags, sorted from [0] oldest to [-1] newest.
+
+    PreConditions:
+    - the git cli tool needs to be installed
+    - the git cli tool can be found via $PATH
+    - the code is executed where the working directory is within a git repository
+    """
+    command = ["git", "tag", "--sort=committerdate"]
+    result = run(command, stdout=PIPE, stderr=PIPE, check=True)
+    tags = (tag.strip() for tag in result.stdout.decode("utf-8").splitlines())
+    return list(tags)
+
+
+def _current_branch():
+    """
+    Returns the current branch name.
+
+    PreConditions:
+    - the git cli tool needs to be installed
+    - the git cli tool can be found via $PATH
+    - the code is executed where the working directory is within a git repository
+    """
+    command = ["git", "branch", "--show-current"]
+    result = run(command, stdout=PIPE, stderr=PIPE, check=True)
+    branch_name = result.stdout.decode("utf-8").strip()
+    return branch_name
+
+
+@nox.session(python=False)
+def clean(session):
+    """Remove all build and documentation artifacts"""
+    build_directory = DOC / "build"
+    if build_directory.exists():
+        rmtree(build_directory.resolve())
+        session.log(f"Removed {build_directory}")
+
+
+@nox.session(python=False)
+def build(session: nox.session):
+    """Build the documentation"""
+    with session.chdir(DOC):
+        session.run(
+            "sphinx-apidoc",
+            "-T",
+            "-e",
+            "-o",
+            "api",
+            f"{PACKAGE.resolve()}",
+            external=True,
+        )
+        session.run("sphinx-build", "-b", "html", "-W", ".", "build", external=True)
+
+
+@nox.session(python=False)
+def open(session):
+    """Open the documentation in the browser"""
+    index_page = DOC / "build" / "index.html"
+    if not index_page.exists():
+        session.error(
+            (f"File {index_page} does not exist." "Please run `nox -s build` first")
+        )
+    session.run(
+        "python", "-m", "webbrowser", "-t", f"{index_page.resolve()}", external=True
+    )
+
+
+@nox.session(python=False)
+def tests(session):
+    """Run all unit tests"""
+    session.run("pytest", f"{PROJECT_ROOT / 'tests'}")
+
+
+@nox.session(python=False)
+@nox.parametrize("branch", ["current", "main"])
```

Comment:
I am not sure, if we need this parameterized

Reply:
I think it it reasonable because the 2 scripts it replaces only vary in a few parameters

Reply:
ah, I missed something, you already restricted the values to current and main, I thought, the branch would be freely choosable. Then it should be fine.